### PR TITLE
Feat/filter tasks by status

### PR DESCRIPTION
**Pull request by niaYaro**

### CODE REVIEW

File: js/show-options.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        const currentBox = editBoxArray.find(function (item) {
+            return id === item.dataset.id
+        })
+        currentBox.classList.toggle('hidden');
+    }
+    const inputTitleList = document.querySelectorAll('.edit__box-title');
+const inputStartDateList = document.querySelectorAll('.timeframe__start');
+const inputEndDateList = document.querySelectorAll('.timeframe__end');
+const changeDataButtons = document.querySelectorAll('.change__button');
+changeDataButtons.forEach(activeButton => activeButton.addEventListener('click', saveData));
```
Comment (vitrovyi) [inline js/show-options.js:28]: @niaYaro
Please refactor the code in the whole file.
Now you are using a lot of selectors and listeners. This is not good.
Please cover your inputs by

tag.
And get your input values from the FormData object.
Then manipulate with gathered data.
Some useful links:
https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData
https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects
https://code-boxx.com/get-html-form-data-javascript/

**Review by vitrovyi — CHANGES_REQUESTED** [review]

File: js/checkbox.js [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
 
-    // This is value of the checkbox
     const checked = event.target.checked;
-    console.log('checked - ', checked);
     const currentElement = tasks.find(function (item) {
         return item.id == id;
     })
-    console.log('task object - ', currentElement);
-    console.log('task status - ', currentElement.status);
+    console.log('currentelement', currentElement)
```
Comment (vitrovyi) [inline js/checkbox.js:8]: @niaYaro 
you forget this one )

File: js/new-cycle.js [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 
 function createCycles() {
     newCecleButton.classList.remove('inactive');
 }
 function newTask() {
-    tasks.push(Array.from(document.querySelectorAll('#task-form input')).reduce((acc, input) => ({...acc, [input.id]: input.value }), {}))
+    tasks.push(Array.from(document.querySelectorAll('#task-form input')).
+    reduce((acc, input) => ({...acc, [input.id]: input.value, [input.id]: input.value }), {}))
     newCecleButton.classList.add('inactive');
+    console.log('tasks', tasks)
```
Comment (vitrovyi) [inline js/new-cycle.js:11]: remove logs in this file

File: js/planning-cycle.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 function showCycles() {
     const planningCyclesContainer = document.getElementById('main__planning-cycles-container');
     planningCyclesContainer.classList.remove('inactive');
+    // console.log('get', localStorage.getItem("allTasks"));
```
Comment (vitrovyi) [inline js/planning-cycle.js:4]: remove commented code

File: js/save-data.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const saveButton = document.querySelector('.storage__wrapper-button');
+saveButton.addEventListener('click', saveDataInLocalStorage);
+function saveDataInLocalStorage () {
+    localStorage.setItem("allTasks", JSON.stringify(tasks))
+}
+function takeDataInLocalStorage() {
+    const savedData = JSON.parse(localStorage.getItem('allTasks'));
+    tasks = savedData;
```
Comment (vitrovyi) [inline js/save-data.js:8]: @niaYaro
We have bug here. If user don't have data in localStorage he can not render tasks.

Please refactore this:
```    if (tasks && tasks.length > 0) {
        taskContainer.innerHTML = '';
        renderTast(tasks);
    }```

File: js/show-options-draft.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+function addEventListenerToPopUp() {
+    const optionButton = document.querySelectorAll('.titlebox__options');
+    const popUpOfOptions = document.querySelectorAll('.popup');
+    optionButton.forEach(currentOptionButton => currentOptionButton.addEventListener('click', showPopUp));
+    function showPopUp(event) {
```
Comment (vitrovyi) [inline js/show-options-draft.js:5]: Please add empty lines between functions and other code.

**Review by vitrovyi — CHANGES_REQUESTED** [review]
Good job!
Please go through comments and make small changes.

File: js/show-options.js (hunk 1 above)
Comment (niaYaro) [reply js/show-options.js]: done

File: js/checkbox.js (hunk 2 above)
Comment (niaYaro) [reply js/checkbox.js]: done

File: js/new-cycle.js (hunk 3 above)
Comment (niaYaro) [reply js/new-cycle.js]: done

File: js/planning-cycle.js (hunk 4 above)
Comment (niaYaro) [reply js/planning-cycle.js]: removed

File: js/save-data.js (hunk 5 above)
Comment (niaYaro) [reply js/save-data.js:10]: done

File: js/show-options-draft.js (hunk 6 above)
Comment (niaYaro) [reply js/show-options-draft.js:6]: added empty lines between functions

**Comment by niaYaro** [comment]
Made correction guided with your comments

**Review by vitrovyi — APPROVED** [review]
@niaYaro 
Good job!
You can merge!

**Pull request merged**